modules/core/shared/src/main/scala/net/protocol/ParseDescribe.scala

```diff
@@ -70,17 +74,17 @@ object ParseDescribe {
                 _  <- parseCache.value.put(stmt, id)
               } yield ()
 
-              (pre, post)
+              (pre, post, closeEvictedStatements)
             }
           case Left(err) =>
-            UnknownTypeException(stmt, err, ty.strategy).raiseError[F, (F[StatementId], StatementId => F[Unit])]
+            UnknownTypeException(stmt, err, ty.strategy).raiseError[F, (F[StatementId], StatementId => F[Unit], F[Unit])]
 
         }
 
       override def command[A](cmd: skunk.Command[A], ty: Typer): F[StatementId] = {
 
         def describeExchange(span: Span[F]): F[(StatementId => F[Unit], F[Unit])] = 
-          OptionT(cache.commandCache.get(cmd)).as(((_: StatementId) => ().pure, ().pure)).getOrElse {
+          OptionT(cache.commandCache.get(cmd)).as(((_: StatementId) => ().pure, cache.commandCache.clearEvicted.void)).getOrElse {
```

Comment:
Have to clear evicted cached entries for command cache here (and query cache below) just to ensure we don't leak entries. Unlike statements, we don't need to take any action on the evicted entries -- just ensure they are removed from the eviction space in the underlying `SemispaceCache`.